src/actix/auth.rs

```diff
@@ -126,33 +126,46 @@ where
             return Box::pin(self.service.call(req));
         }
 
-        // Grab API key from request
-        let key =
-            // Request header
-            req.headers().get("api-key").and_then(|key| key.to_str().ok()).map(|key| key.to_string())
-                // Fall back to authentication header with bearer token
-                .or_else(|| {
-                    Authorization::<Bearer>::parse(&req).ok().map(|auth| auth.as_ref().token().into())
-                });
-
-        if let Some(key) = key {
-            let is_allowed = if let Some(ref auth_keys) = self.auth_keys {
-                auth_keys.can_write(&key) || (is_read_only(&req) && auth_keys.can_read(&key))
-            } else {
-                // This code path should not be reached
-                log::warn!("Auth for REST API is set up incorrectly. Denying access by default.");
-                false
-            };
-            if is_allowed {
+        let invalid_api_key = |req: ServiceRequest| -> Self::Future {
+            Box::pin(async {
+                Ok(req
+                    .into_response(HttpResponse::Forbidden().body("Invalid api-key"))
+                    .map_into_right_body())
+            })
+        };
+
+        let Some(ref auth_keys) = self.auth_keys else {
+            // This code path should not be reached
+            log::warn!("Auth for REST API is set up incorrectly. Denying access by default.");
+            return invalid_api_key(req);
+        };
```

Comment:
I know this was here before, but we can fix this by setting `auth_keys: AuthKeys` in the definition (without `Option`). The constructor should only work if we have indeed AuthKeys